### PR TITLE
add a note regarding feature detection

### LINKED ISSUE #64: Add a note regarding feature detection options for new entry types
While reviewing a LongTask implementation [patch](https://codereview.chromium.org/2610063006/) I noticed that there's no way for developers to detect browser support for new entry types:

* In [Filter performance entry buffer by name and type](https://w3c.github.io/performance-timeline/#filter-performance-entry-buffer-by-name-and-type), we just return an empty list for unknown types.
* In [The PerformanceObserver interface](https://w3c.github.io/performance-timeline/#the-performanceobserver-interface), we specify "types not recognized by the user agent must be ignored" and the processing model later filters and ignores unsupported types.

I think we should change that so that at least one but ideally both throw or return an error when called with an unsupported entry type.

/cc @spanicker @igrigorik

### PR DESCRIPTION
**Pull request by igrigorik**
closes #64.

@yoavweiss ptal.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -400,6 +400,12 @@
           entryType names</a> within the `entryTypes` sequence, and replace
           the `entryTypes` sequence with the new filtered sequence.
         </li>
+        <p class="note">In some cases the developer may want to distinguish
```
Comment (yoavweiss) [inline index.html]: It might be better to add the note outside of the processing model. (so right after the `</ol>`)

File: index.html [hunk 2]
```diff
@@ -400,6 +400,12 @@
...
+        between absence and lack of support for a particular type of entry. In
```
Comment (yoavweiss) [inline index.html]: That first phrase is not very clear and doesn't cover polyfills. Maybe "In some cases developers may want to distinguish between the absence of an entry in the timeline and lack of support for that entry, or may want to conditionally create alternative measurements for unsupported entry types."

File: index.html [hunk 3]
```diff
@@ -400,6 +400,12 @@
...
+        such cases they can feature detect the presense of the relevant
```
Comment (yoavweiss) [inline index.html]: s/feature detect/feature-detect/
s/presense/presence/

File: index.html [hunk 4]
```diff
@@ -400,6 +400,12 @@
...
+        interface—e.g. to detect if User Timing entries are supported, the
```
Comment (yoavweiss) [inline index.html]: IIUC we're testing on the existence of the constructor, not the interface.

**Comment by igrigorik** [comment]
@yoavweiss thanks, updated.. ptal.

**Comment by yoavweiss** [comment]
LGTM

**Pull request merged**